feat: Monster hit

leejung7308 opened the pull request:
몬스터 피격:
- 원거리: 충돌 오브젝트 삭제 후 피격 로그 출력
- 근거리: 피격 로그 출력

버그 수정:
원거리 무기가 상대 오브젝트와 충돌 시 삭제 되도록 설계해서 두 오브젝트가 가까이 붙었을 때 손에 들고 있는 오브젝트까지 삭제시키는 버그가 발생
- 손에 들고 있는 무기의 콜라이더를 비활성화 시키고 발사되는 투사체 오브젝트만 콜라이더를 활성화하는 방법으로 해결

leejung7308 committed:
feat: Add weapon tag

Melee Weapon(Player)
Ranged Weapon(Player)
Melee Weapon(Monster)
Ranged Weapon(Monster)

leejung7308 committed:
feat: Update prefab tag

leejung7308 committed:
feat: Update Scene

몬스터 다중 콜라이더 처리

leejung7308 committed:
feat: Update Player script

collider tag 및 로그 출력문 수정

leejung7308 committed:
feat: Monster Hit

원거리 무기 : 오브젝트 삭제(Destroy) 후 로그 생성
근거리 무기 : 로그 생성

아이템 추가 후 데미지 및 애니메이션 추가 예정

leejung7308 committed:
feat: Update Scene

몬스터 오브젝트 스크립트 옵션 변경

leejung7308 committed:
feat: Fix Monster collider bug

콜라이더 버그 수정
손에 든 원거리 무기는 콜라이더 off
발사되는 원거리 무기만 콜라이더 on

leejung7308 committed:
feat: Fix Player collider bug

콜라이더 버그 수정
손에 든 원거리 무기는 콜라이더 off
발사되는 원거리 무기만 콜라이더 on

Lapis0875 commented on Assets/Scripts/Player.cs:
```diff
@@ -69,6 +69,7 @@ void WeaponSwap()
             weapons[2].gameObject.SetActive(false);
             weapons[3].gameObject.SetActive(false);
             weapon = weapons[1];
+            weapon.GetComponent<Collider2D>().enabled = false;
```
무기 스왑하는 부분들 겹치는 코드가 꽤 많은데, 이거 별도 메소드로 분리해서 코드 재사용성 높이면 어떨까?

leejung7308 replied:
고쳤습니다

Lapis0875 reviewed (changes requested):
수고했어! 몇가지 중복해서 쓰이는 코드들만 모듈화 하면 더 깔끔해질 것 같아.

leejung7308 committed:
feat: Update Scene

플레이어 오브젝트 하위 Weapons 오브젝트 생성: 초기에 생성되는 무기 오브젝트 담는 용도

leejung7308 committed:
feat: Update Player script

- 초기에 생성되는 무기들이 플레이어 오브젝트의 하위 오브젝트로 들어가도록 함
- bool 타입 isMelee 변수를 추가해 손에 들고 있는 무기의 콜라이더 활성화/비활성화 여부를 결정
- Shoot() 추가
- SetWeapon() 추가:  SwapWeapon 간소화

leejung7308 committed:
feat: Update Monster script

몬스터의 무기가 몬스터 오브젝트의 하위 오브젝트로 생성되도록 함

20192094 reviewed (approved):
ㅇㅋㅇㅋ

Lapis0875 reviewed (approved):
수고했어~
